### PR TITLE
RMSW-78

### PR DESCRIPTION
**Pull request by tiffcmw**
Added code for converting and storing ADC values from analog sensors. 

Added the following GPIO PINS in .ioc:
PA3: USART2_RX
PD5: USART2_TX
PA4: ADCI_IN4 (Channel 4)
PA5: ADC1_IN5 (Channel 5)
PC8: GPIO_Output
PC9: GPIO_Output
PC10: GPIO_Input
PC11: GPIO_Input

Key Changes: 
1. Circular Buffer Implementation:
- A circular buffer (adc_circular_buffer) is used to store ADC samples.
- Functions like addSampleToBuffer, getLatestSample, and getRecentSamples manage this buffer.

2. ADC Data Processing:
- The ProcessADCData function processes ADC data for channels 4 and 5 (PA4 and PA5 respectively).
- It stores the data in the circular buffer and optionally sends debug information via UART.

3. Data Retrieval:
- getAnalogSensorData retrieves the latest data for a specific sensor (currently only supports channels 4 and 5).

The code has only been tested in a private project. Further testing has to be made depending on the actual input and output sources in the car's system.

### CODE REVIEW

File: Core/Src/main.c [hunk 1]
```diff
@@ -384,6 +506,9 @@ static void MX_GPIO_Init(void)
   /*Configure GPIO pin Output Level */
   HAL_GPIO_WritePin(USB_PowerSwitchOn_GPIO_Port, USB_PowerSwitchOn_Pin, GPIO_PIN_RESET);
 
+  /*Configure GPIO pin Output Level */
```
Comment (AmarMaksumic) [inline Core/Src/main.c:509]: Remove this so it doesnt confuse others

**Review by AmarMaksumic — CHANGES_REQUESTED** [review]
Code is very good! Let's just implement all the channels now and do some cleanup with variable names and logic~ Thank you!

File: Core/Src/Sensors/AnalogSensor.c [hunk 2]
```diff
@@ -131,9 +147,87 @@ void initAnalogSensor(AnalogSensor* sensor, const char* name, int hz, int channe
...
+
+/**
+ * @brief Get the latest ADC sample from the buffer
+ * @return ADCSample The latest ADC sample
+ */
+ADCSample getLatestSample(void) {
+    if (buffer_count > 0) {
+        uint32_t latest = (buffer_head - 1 + BUFFER_SIZE) % BUFFER_SIZE;
+        return adc_circular_buffer[latest];
+    } else {
```
Comment (Arct1cPharaoh) [inline Core/Src/Sensors/AnalogSensor.c:197]: Since you return in the if statement you could just not have an else statement.

**Review by Arct1cPharaoh — CHANGES_REQUESTED** [review]
One small detail. Other than that lgtm

**Review by AmarMaksumic — COMMENTED** [review]
LGTM, only question is did you test all three ADCs outside of this code?

**Review by Arct1cPharaoh — APPROVED** [review]
lgtm. Good work!

**Review by AmarMaksumic — APPROVED** [review]
Tiffany tested code, made sure all three ADCs can work at same time. Approved!

**Pull request merged**